pandas/core/groupby/grouper.py

```diff
@@ -542,6 +537,14 @@ def __init__(
                 # TODO 2022-10-08 we only have one test that gets here and
                 #  values are already in nanoseconds in that case.
                 self.grouping_vector = Series(self.grouping_vector).to_numpy()
+        elif is_categorical_dtype(self.grouping_vector):
```

Comment:
any particular reason this was moved from above?

Reply:
Previously, this block was in an `if...elif...elif...` chain where it would be skipped over when the first if was true (namely, when the grouper is specified by a level in the index). Now it's moved out of that chain, so it's always hit.